ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -497,27 +494,31 @@ def test_elementwise_udf_overwrite_destruct_and_assign(
 
 
 @pytest.mark.min_spark_version('3.1')
-def test_elementwise_udf_destruct_exact_once(udf_backend, udf_alltypes):
-    with tempfile.TemporaryDirectory() as tempdir:
-
-        @elementwise(
-            input_type=[dt.double],
-            output_type=dt.Struct(['col1', 'col2'], [dt.double, dt.double]),
-        )
-        def add_one_struct_exact_once(v):
-            key = v.iloc[0]
-            path = Path(f"{tempdir}/{key}")
-            assert not path.exists()
-            path.touch()
-            return v + 1, v + 2
-
-        result = udf_alltypes.mutate(
-            add_one_struct_exact_once(udf_alltypes['index']).destructure()
+def test_elementwise_udf_destruct_exact_once(
+    udf_backend, udf_alltypes, tmp_path
+):
+    if udf_backend.name() in ['pandas', 'dask']:
+        pytest.skip(
```

Comment:
Yes, that's the actual breaking change.

Reply:
I remembered that pandas does have "exact once" semantics for destructure UDFs before. Is it that true?

Reply:
If we were about to merge this PR, then yes. The behavior could be still still restored by refactoring the pandas execution to reuse/cache results of the expressions (with additional speedups).

Reply:
> If we were about to merge this PR, then yes.

Just to confirm:
Before the PR - we have exact once for pandas
After the PR - we don't have exact once for pandas
?

Reply:
Correct.

Reply:
This is going to be an issue for us because internally we use this feature pretty heavily and would be a perf regression if we started to execute this types of UDF multiple times...

Reply:
Thanks for checking! I'll try to find a solution to keep the previous behavior then.

Reply:
To give a bit more context, during previous work. I think we tried previous to destructure the assignment directly (similar to what you are doing) but @emilyreff noticed that Pandas backend cannot share intermediate result among different struct field object that points to the same struct and opened https://github.com/ibis-project/ibis/issues/2649

And I believe this is the reason that the code is written in the current way... (with a destructure expr)

Reply:
(Sorry should probably have linked the issue in code comments to be clear)